lua/lsp/handlers.lua

```diff
@@ -60,4 +43,64 @@ function M.setup()
   })
 end
 
+function M.show_line_diagnostics()
+  local diagnostics = vim.lsp.diagnostic.get_line_diagnostics()
+  local diags = vim.deepcopy(diagnostics)
+  local height = #diagnostics
+  local width = 0
+  local opts = {}
+  local close_events = { "CursorMoved", "CursorMovedI", "BufHidden", "InsertCharPre" }
+  local bufnr = vim.api.nvim_create_buf(false, true)
+  local diagnostic_severities = {
+    "Error",
+    "Warning",
+    "Information",
+    "Hint",
+  }
+
+  for i, v in ipairs(diagnostics) do
+    local source = v.source
+    if source then
+      if string.find(source, "/") then
+        source = string.sub(v.source, string.find(v.source, "([%w-_]+)$"))
+      end
+      diags[i].message = string.format("%s: %s", source, v.message)
+    else
+      diags[i].message = string.format("%s", v.message)
+    end
+
+    if vim.tbl_contains(vim.tbl_keys(v), "code") then
+      diags[i].message = diags[i].message .. string.format(" [%s]", v.code)
```

Comment:
A detail, but I would write it `string.format("%s [%s]", diags[i].message, v.code)`
I like to see the complete format and not a mix of concatenated strings :)